tasks/compile.rake

```diff
@@ -77,7 +81,7 @@ task :devkit do
 end
 
 if RUBY_PLATFORM =~ /mingw|mswin/
-  Rake::Task['compile'].prerequisites.unshift 'vendor:mysql'
+  Rake::Task['compile'].prerequisites.unshift 'vendor:mysql' unless defined?(RubyInstaller)
```

Comment:
`rake spec` depends on `compile`, which is OK, but it shouldn't depend on the vendored mysql library on RubyInstaller-2.4.

The reason is: the simplest way to build the mysql2 gem is by using the `libmariadbclient` pacman package. It is automatically installed though the `metadata['msys2_mingw_dependencies']` tag when running `gem install mysql2`

On the other hand the vendored library works as well, provided that the `unzip` command is installed. This can be done per `pacman -S unzip`. But I think that this shouldn't the default way on RubyInstaller-2.4.